mleap-spark/src/main/scala/org/apache/spark/ml/bundle/ops/feature/BinarizerOp.scala

```diff
@@ -22,25 +24,36 @@ class BinarizerOp extends SimpleSparkOp[Binarizer] {
       assert(context.context.dataset.isDefined, BundleHelper.sampleDataframeMessage(klazz))
 
       val dataset = context.context.dataset.get
-      model.withValue("threshold", Value.double(obj.getThreshold)).
-        withValue("input_shapes", Value.dataShape(sparkToMleapDataShape(dataset.schema(obj.getInputCol), dataset)))
+      var result = {
+        ParamValidators.checkSingleVsMultiColumnParams(obj, Seq(obj.inputCol), Seq(obj.inputCols))
+        if(obj.isSet(obj.inputCols)) {
+          model.withValue("input_shapes", Value.dataShape(ListShape()))
+        } else {
+          model.withValue("input_shapes", Value.dataShape(sparkToMleapDataShape(dataset.schema(obj.getInputCol), dataset)))
+        }
+      }
+
+      if (obj.isSet(obj.threshold)) result = result.withValue("threshold", Value.double(obj.getThreshold))
+      if (obj.isSet(obj.thresholds)) result = result.withValue("thresholds", Value.doubleList(obj.getThresholds))
+      saveMultiInOutFormat(result, obj)
     }
 
     override def load(model: Model)
                      (implicit context: BundleContext[SparkBundleContext]): Binarizer = {
-      new Binarizer(uid = "").setThreshold(model.value("threshold").getDouble)
+      val threshold: Option[Double] = model.getValue("threshold").map(_.getDouble)
+      val thresholds: Option[Seq[Double]] = model.getValue("thresholds").map(_.getDoubleList)
+      val binarizer = new Binarizer()
+      val obj = (threshold, thresholds) match {
+        case (None, None) => throw new IllegalArgumentException("Neither threshold nor thresholds were found")
+        case (Some(v), None) => binarizer.setThreshold(v)
+        case (None, Some(v)) => binarizer.setThresholds(v.toArray)
+        case (_, _) => throw new IllegalArgumentException("Both thresholds and threshold were found")
+      }
+      loadMultiInOutFormat(model, obj)
     }
   }
 
   override def sparkLoad(uid: String, shape: NodeShape, model: Binarizer): Binarizer = {
-    new Binarizer(uid = uid).setThreshold(model.getThreshold)
-  }
-
-  override def sparkInputs(obj: Binarizer): Seq[ParamSpec] = {
-    Seq("input" -> obj.inputCol)
-  }
-
-  override def sparkOutputs(obj: Binarizer): Seq[SimpleParamSpec] = {
-    Seq("output" -> obj.outputCol)
+    model
```

Comment:
same as above, should it do `new Binarizer(uid = uid).setThreshold(model.getThreshold)`?